### PR TITLE
feat: persist view filters and sorts on Update View button click

### LINKED ISSUE #1121: Create View Filter (field, operand, value, view_id)
Create `ViewFilter` prisma model.

### LINKED ISSUE #1123: On companies/people page when I update filters, they are persisted
```[tasklist]
### Tasks
- [x] Add ViewFilter resolvers: findManyViewFilter, createManyViewFilter, updateOneViewFilter, deleteManyViewFilter
- [x] Add a useViewFilters hook that loads the filters and returns a persistFilters method, which can be called on view submit
```

### LINKED ISSUE #1124: Add an "Update view" button to persist the filters

### LINKED ISSUE #1289: Add an arrow dropdown beside "Update view" button with a "Create View" option

### PR DESCRIPTION
**Pull request by thaisguigon**
Closes #1121
Closes #1123 
Closes #1124 
Closes #1289

### CODE REVIEW

File: front/src/modules/ui/filter-n-sort/states/savedFiltersByKeyScopedState.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { selectorFamily } from 'recoil';
+
+import type { Filter } from '../types/Filter';
+
+import { savedFiltersScopedState } from './savedFiltersScopedState';
+
+export const savedFiltersByKeyScopedState = selectorFamily({
```
Comment (emilienchvt) [inline front/src/modules/ui/filter-n-sort/states/savedFiltersByKeyScopedState.ts:7]: Nice!

File: front/src/modules/ui/filter-n-sort/states/savedFiltersByKeyScopedState.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  key: 'savedFiltersByKeyScopedState',
```
Comment (emilienchvt) [inline front/src/modules/ui/filter-n-sort/states/savedFiltersByKeyScopedState.ts:8]:
```suggestion
  key: 'savedFiltersByKeyScopedSelector',
```

File: front/src/modules/ui/filter-n-sort/states/savedSortsByKeyScopedState.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { selectorFamily } from 'recoil';
+
+import type { SelectedSortType } from '../types/interface';
+
+import { savedSortsScopedState } from './savedSortsScopedState';
+
+export const savedSortsByKeyScopedState = selectorFamily({
+  key: 'savedSortsByKeyScopedState',
```
Comment (emilienchvt) [inline front/src/modules/ui/filter-n-sort/states/savedSortsByKeyScopedState.ts:8]:
```suggestion
  key: 'savedSortsByKeyScopedSelector',
```

File: front/src/modules/ui/filter-n-sort/states/sortsOrderByScopedState.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { selectorFamily } from 'recoil';
+
+import { reduceSortsToOrderBy } from '../helpers';
+
+import { sortsScopedState } from './sortsScopedState';
+
+export const sortsOrderByScopedState = selectorFamily({
+  key: 'sortsOrderByScopedState',
```
Comment (emilienchvt) [inline front/src/modules/ui/filter-n-sort/states/sortsOrderByScopedState.ts:8]:
```suggestion
  key: 'sortsOrderByScopedSelector',
```

File: front/src/modules/views/hooks/useViewFilters.ts [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+      const nextFilters = data.viewFilters
+        .map((viewFilter) => {
+          const availableFilter = availableFilters.find(
+            (filter) => filter.field === viewFilter.key,
+          );
+
+          return availableFilter
+            ? ({
+                displayValue: viewFilter.displayValue ?? viewFilter.value,
+                field: viewFilter.key,
```
Comment (emilienchvt) [inline front/src/modules/views/hooks/useViewFilters.ts:58]: Should we have back and front agree on a naming `key` or `field` ?

**Review by emilienchvt — COMMENTED** [review]
Nice one! A few quick comments on the naming of selectors and vocabulary diff between front and back.

When testing the feature I was surprised by a few behaviours:

- When creating a new view, I’d expect it to be selected
- After I create a filter on a new view, clicking on the “Filter button” toggles the filter&sorts bar which I find weird (Should enable new filter creation instead right ?)
- No feedback when I “Update view” => I’d expect the button to be disabled **Edit:** addressed with https://github.com/twentyhq/twenty/pull/1293#pullrequestreview-1591224772
- When a view is selected, I can see a “cancel” button. I find it unclear what it does: it removes the filters and sorts of the view but does not change the “selected view” in the top bar.

File: front/src/modules/ui/filter-n-sort/states/savedFiltersByKeyScopedState.ts (hunk 2 above)
Comment (thaisguigon) [reply front/src/modules/ui/filter-n-sort/states/savedFiltersByKeyScopedState.ts]: Done, thank you!

**Comment by charlesBochet** [comment]
> * one, it might be addressed in the upcoming filters & sorts bar refactoring as well. Maybe labelling it differently ("Discard" for instance?) would make it clear

I think we should rediscuss this once we have cleared the behaviors regarding the sort and filter chips bar. What opens it, what closes it? Then we can discuss Cancel button role. I would consider it separate from this PR

**Review by charlesBochet — APPROVED** [review]
Great!

**Pull request merged**